presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -691,32 +693,39 @@ private static StorageDescriptor makeStorageDescriptor(String tableName, List<Co
     {
         TypeInfo typeInfo = columnType.getTypeInfo();
-        checkArgument(typeInfo.getCategory() == PRIMITIVE, "unsupported type: %s", columnType);
-        switch (((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory()) {
-            case BOOLEAN:
-                return createBooleanStatistics(columnName, columnType, statistics);
-            case BYTE:
-            case SHORT:
-            case INT:
-            case LONG:
-                return createLongStatistics(columnName, columnType, statistics);
-            case FLOAT:
-            case DOUBLE:
-                return createDoubleStatistics(columnName, columnType, statistics);
-            case STRING:
-            case VARCHAR:
-            case CHAR:
-                return createStringStatistics(columnName, columnType, statistics, rowCount);
-            case DATE:
-                return createDateStatistics(columnName, columnType, statistics);
-            case TIMESTAMP:
-                return createLongStatistics(columnName, columnType, statistics);
-            case BINARY:
-                return createBinaryStatistics(columnName, columnType, statistics, rowCount);
-            case DECIMAL:
-                return createDecimalStatistics(columnName, columnType, statistics);
-            default:
-                throw new IllegalArgumentException(format("unsupported type: %s", columnType));
+        if (typeInfo.getCategory() == PRIMITIVE) {
+            switch (((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory()) {
+                case BOOLEAN:
+                    return createBooleanStatistics(columnName, columnType, statistics);
+                case BYTE:
+                case SHORT:
+                case INT:
+                case LONG:
+                    return createLongStatistics(columnName, columnType, statistics);
+                case FLOAT:
+                case DOUBLE:
+                    return createDoubleStatistics(columnName, columnType, statistics);
+                case STRING:
+                case VARCHAR:
+                case CHAR:
+                    return createStringStatistics(columnName, columnType, statistics, rowCount);
+                case DATE:
+                    return createDateStatistics(columnName, columnType, statistics);
+                case TIMESTAMP:
+                    return createLongStatistics(columnName, columnType, statistics);
+                case BINARY:
+                    return createBinaryStatistics(columnName, columnType, statistics, rowCount);
+                case DECIMAL:
+                    return createDecimalStatistics(columnName, columnType, statistics);
+                default:
+                    throw new IllegalArgumentException(format("unsupported type: %s", columnType));
+            }
+        }
+        else if (typeInfo.getCategory() == STRUCT || typeInfo.getCategory() == MAP) {
+            return new ColumnStatisticsObj();
```

Comment:
It looks like we only return an empty stats object now. I'm curious if we will be filling up the actual stats for structs and maps? How does the `analyze` result look like now with this change? Does it just leave that part empty? Is it possible to attach a printout of the result of it here?

Reply:
yes, that's correct we only return an empty stats right now since this is a fix for issue #14494. Here's a example of the output with and without this fix
```
presto:sj_test> select * from test;
     a      | b | c 
------------+---+---
 {b=1, c=2} | 3 | a 
(1 row)
```
without this change:
```
presto:sj_test> analyze test;
ANALYZE: 1 row

Query 20200515_162351_00021_y4gxt, FAILED, 2 nodes
Splits: 11 total, 11 done (100.00%)
0:00 [1 rows, 450B] [5 rows/s, 2.5KB/s]

Query 20200515_162351_00021_y4gxt failed: unsupported type: struct<b:int,c:int>
```

with this change:
```
presto:sj_test> analyze test;
ANALYZE: 1 row

Query 20200515_161538_00024_84q6a, FINISHED, 3 nodes
Splits: 11 total, 11 done (100.00%)
0:00 [1 rows, 452B] [10 rows/s, 4.8KB/s]
```

Reply:
I agree with @yingsu00  here. I can't tell if the hive metastore accepts column stats for struct types (for the non type specific fields), but if it does and we have the stats here, I think you should create a proper stats object here with those things set 

Then you can test what happened by running "show stats" after running "analyze"

Reply:
@sujay-jain @yingsu00 @rschlussel It seems like ANALYZE statement never collected statistics for complex types, hence, fixing ANALYZE to not fail and produce empty statistics is a reasonable change. We may want to create an issue to add support for collecting stats for complex types, but priority of this would not be high and this can be handled separately at some future time.

Reply:
Sure, but this just seems like such a quick thing to test out that I feel like it makes sense to try it now

Reply:
Also @sujay-jain  can you show the show stats output for the current implementation.  Just to confirm the behavior is the same.

Reply:
Thanks @sujay-jain . I'm ok with collecting the actual stats in a separate task/issue.

Reply:
> I agree with @yingsu00 here. I can't tell if the hive metastore accepts column stats for struct types (for the non type specific fields), but if it does and we have the stats here, I think you should create a proper stats object here with those things set

I don't see Struct or Map type ```ColumnStatsData``` in ```org.apache.hadoop.hive.metastore.api``` like there are ```ColumnStatsData``` types for primitives eg: ```BooleanColumnStatsData```. As far as I can tell this is not supported by default - we may have to implement something on our own. 
@rschlussel  let me know your thoughts regarding this and we can create a task/issue for it.

Reply:
Here's an output for ```analyze``` and ```show stats``` with my current change
```
presto:sj_test> analyze test;
ANALYZE: 1 row

Query 20200519_055255_00001_xtxwu, FINISHED, 1 node
Splits: 35 total, 35 done (100.00%)
0:01 [1 rows, 464B] [0 rows/s, 392B/s]
```
```
presto:sj_test> show stats for test;
 column_name | data_size | distinct_values_count | nulls_fraction | row_count | low_value | high_value 
-------------+-----------+-----------------------+----------------+-----------+-----------+------------
 a           | NULL      | NULL                  | NULL           | NULL      | NULL      | NULL       
 b           | NULL      |                   1.0 |            0.0 | NULL      | 3         | 3          
 c           |       1.0 |                   1.0 |            0.0 | NULL      | NULL      | NULL       
 NULL        | NULL      | NULL                  | NULL           |       1.0 | NULL      | NULL       
(4 rows)

Query 20200519_055302_00002_xtxwu, FINISHED, 1 node
Splits: 1 total, 1 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]
```

Reply:
@sujay-jain nevermind then.  It was just an opportunisitic suggestion if it was easy.